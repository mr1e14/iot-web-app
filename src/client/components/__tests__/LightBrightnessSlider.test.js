import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import LightBrightnessSlider from "../LightBrigtnessSlider";

const propsEnabled = {
  on: true,
  connected: true,
  brightness: 25
};

const propsDisabled = {
  on: false,
  connected: true,
  brightness: 25
};

const propsDisconnected = {
  on: undefined,
  connected: false,
  brightness: undefined
};

describe("LightBrightnessSlider", () => {
  let tree;
  let component;
  let wrapper;
  describe("when light is enabled", () => {
    beforeEach(() => {
      component = renderer.create(<LightBrightnessSlider {...propsEnabled} />);
      tree = component.toJSON();
      wrapper = mount(<LightBrightnessSlider {...propsEnabled} />);
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
    it("is enabled", () => {
      expect(wrapper.find("Slider").prop("disabled")).toBe(false);
    });
    it("may be used to adjust brightness", () => {
      const newBrightness = 75;
      wrapper
        .find("Slider")
        .props()
        .onChange({}, newBrightness);
      expect(wrapper.find("LightBrightnessSlider").state()).toEqual({
        value: newBrightness
      });
    });
  });
  describe("when light is disabled", () => {
    beforeEach(() => {
      component = renderer.create(<LightBrightnessSlider {...propsDisabled} />);
      tree = component.toJSON();
      wrapper = mount(<LightBrightnessSlider {...propsDisabled} />);
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
    it("is disabled", () => {
      expect(wrapper.find("Slider").prop("disabled")).toBe(true);
    });
  });
  describe("when light is disconnected", () => {
    beforeEach(() => {
      component = renderer.create(
        <LightBrightnessSlider {...propsDisconnected} />
      );
      tree = component.toJSON();
      wrapper = mount(<LightBrightnessSlider {...propsDisconnected} />);
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
    it("is disabled", () => {
      expect(wrapper.find("Slider").prop("disabled")).toBe(true);
    });
  });
});
