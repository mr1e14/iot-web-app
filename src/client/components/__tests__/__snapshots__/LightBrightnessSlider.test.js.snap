// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightBrightnessSlider when light is disabled renders correctly 1`] = `
<div
  aria-disabled={true}
  aria-orientation="horizontal"
  aria-valuemax={100}
  aria-valuemin={1}
  aria-valuenow={25}
  className="MuiSlider-root-4 MuiSlider-disabled-13"
  onClick={[Function]}
  onMouseDown={[Function]}
  onTouchMove={[Function]}
  onTouchStartCapture={[Function]}
  role="slider"
>
  <div
    className="MuiSlider-container-5 LightBrightnessSlider-container-1"
  >
    <div
      className="MuiSlider-track-6 LightBrightnessSlider-track-2 MuiSlider-trackBefore-7 MuiSlider-disabled-13"
      style={
        Object {
          "width": "calc(24.242424242424242% - 6px)",
        }
      }
    />
    <div
      className="MuiSlider-thumbWrapper-9 MuiSlider-disabled-13"
      style={
        Object {
          "transform": "translateX(24.242424242424242%)",
        }
      }
    >
      <button
        className="MuiButtonBase-root-18 MuiButtonBase-disabled-19 MuiSlider-thumb-10 LightBrightnessSlider-thumb-3 MuiSlider-disabled-13"
        disabled={true}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        onTouchStartCapture={[Function]}
        tabIndex="-1"
        type="button"
      />
    </div>
    <div
      className="MuiSlider-track-6 LightBrightnessSlider-track-2 MuiSlider-trackAfter-8 MuiSlider-disabled-13"
      style={
        Object {
          "width": "calc(75.75757575757575% - 6px)",
        }
      }
    />
  </div>
</div>
`;

exports[`LightBrightnessSlider when light is disconnected renders correctly 1`] = `
<div
  aria-disabled={true}
  aria-orientation="horizontal"
  aria-valuemax={100}
  aria-valuemin={1}
  aria-valuenow={1}
  className="MuiSlider-root-4 MuiSlider-disabled-13"
  onClick={[Function]}
  onMouseDown={[Function]}
  onTouchMove={[Function]}
  onTouchStartCapture={[Function]}
  role="slider"
>
  <div
    className="MuiSlider-container-5 LightBrightnessSlider-container-1"
  >
    <div
      className="MuiSlider-track-6 LightBrightnessSlider-track-2 MuiSlider-trackBefore-7 MuiSlider-disabled-13"
      style={
        Object {
          "width": "calc(0% - 6px)",
        }
      }
    />
    <div
      className="MuiSlider-thumbWrapper-9 MuiSlider-disabled-13"
      style={
        Object {
          "transform": "translateX(0%)",
        }
      }
    >
      <button
        className="MuiButtonBase-root-18 MuiButtonBase-disabled-19 MuiSlider-thumb-10 LightBrightnessSlider-thumb-3 MuiSlider-disabled-13"
        disabled={true}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        onTouchStartCapture={[Function]}
        tabIndex="-1"
        type="button"
      />
    </div>
    <div
      className="MuiSlider-track-6 LightBrightnessSlider-track-2 MuiSlider-trackAfter-8 MuiSlider-disabled-13"
      style={
        Object {
          "width": "calc(100% - 6px)",
        }
      }
    />
  </div>
</div>
`;

exports[`LightBrightnessSlider when light is enabled renders correctly 1`] = `
<div
  aria-disabled={false}
  aria-orientation="horizontal"
  aria-valuemax={100}
  aria-valuemin={1}
  aria-valuenow={25}
  className="MuiSlider-root-4"
  onClick={[Function]}
  onMouseDown={[Function]}
  onTouchMove={[Function]}
  onTouchStartCapture={[Function]}
  role="slider"
>
  <div
    className="MuiSlider-container-5 LightBrightnessSlider-container-1"
  >
    <div
      className="MuiSlider-track-6 LightBrightnessSlider-track-2 MuiSlider-trackBefore-7"
      style={
        Object {
          "transform": "translateY(-50%) scaleX(0.24242424242424243)",
        }
      }
    />
    <div
      className="MuiSlider-thumbWrapper-9"
      style={
        Object {
          "transform": "translateX(24.242424242424242%)",
        }
      }
    >
      <button
        className="MuiButtonBase-root-18 MuiSlider-thumb-10 LightBrightnessSlider-thumb-3"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        onTouchStartCapture={[Function]}
        tabIndex="0"
        type="button"
      />
    </div>
    <div
      className="MuiSlider-track-6 LightBrightnessSlider-track-2 MuiSlider-trackAfter-8"
      style={
        Object {
          "transform": "translateY(-50%) scaleX(0.7575757575757575)",
        }
      }
    />
  </div>
</div>
`;
